runtime/src/parachains.rs

```diff
@@ -509,10 +524,11 @@ impl<T: Trait> Module<T> {
 	}
 
 	/// Calculate the current block's duty roster using system's random seed.
-	pub fn calculate_duty_roster() -> DutyRoster {
+	/// Returns the duty roster along with the random seed.
+	pub fn calculate_duty_roster() -> (DutyRoster, [u8; 32]) {
```

Comment:
how about adding a type alias for the random seed instead of `[u8; 32]`?

i.e.
```
pub type RandomSeed = [u8; 32];
```